node/controllers/auth/index.js

```diff
@@ -198,3 +198,37 @@ export const reset = async (req, res) => {
     })
   }
 }
+
+export const checkIfResetUsed = async (req, res) => {
+  const { uuid } = req.body
+
+  try {
+    // Check the token is still valid and if so update 'used' value
+    // If the token is not valid, throw an error
+    const passwordResetEntry = await PasswordResetUuid.findOne({
+      where: { uuid },
+    })
+    if (!passwordResetEntry) {
+      res.status(400).send({
```

Comment:
I just realised that we have more cases like this, but I'm quite sure this should be `return res.status(400).send(...)` since otherwise the function will continue executing and the response will always be 500 because of some other error that occurs later.

Reply:
Other option is to do if-else for these cases

Reply:
Hmm, true! This was actually copied directly from the reset function that first performs this check and then proceeds to reset the password using the data.